docs/dqx/docs/guide/data_contract_quality_rules_generation.mdx

```diff
@@ -0,0 +1,884 @@
+
+Text-based expectations are defined using `type: text` in the quality section:
+
+<Tabs>
+  <TabItem value="YAML" label="Contract (YAML)" default>
+    ```yaml
+    dataContractSpecification: 0.9.3
+    id: urn:datacontract:compliance:user_data
+    info:
+      title: User Data (GDPR Compliant)
+      version: 1.0.0
+
+    models:
+      users:
+        type: table
+        
+        # Dataset-level text expectation
+        quality:
+          - type: text
+            description: |
+              The dataset should not contain duplicate user records based on
+              email address. Users with the same email are considered duplicates
+              and should be flagged for review.
+        
+        fields:
+          email:
+            type: string
+            required: true
+            quality:
+              # Field-level text expectation
+              - type: text
+                description: |
+                  Email addresses must be valid and from approved corporate
+                  domains only (@company.com or @partner.com). Personal email
+                  domains like gmail.com or yahoo.com should be rejected.
+          
+          consent_date:
+            type: date
+            required: true
+            quality:
+              # Another text expectation
+              - type: text
+                description: |
+                  Consent date must not be in the future and should be within
+                  the last 7 years to comply with GDPR retention policies.
+    ```
+  </TabItem>
+</Tabs>
+
+### Generating Rules with Text Expectations
+
+<Tabs>
+  <TabItem value="Python" label="Python" default>
+    ```python
+    from databricks.labs.dqx.profiler.generator import DQGenerator
+    from databricks.sdk import WorkspaceClient
+
+    ws = WorkspaceClient()
+    generator = DQGenerator(workspace_client=ws)
+
+    # Generate rules including text-based expectations (requires LLM)
+    rules = generator.generate_rules_from_contract(
+        contract_file="user_data_contract.yaml",
+        generate_predefined_rules=True,   # Include schema-derived rules
+        process_text_rules=True         # Process text expectations with LLM
+    )
+
+    # The generated rules will include:
+    # - Predefined rules from schema constraints
+    # - Explicit DQX rules (if any)
+    # - Rules generated from text expectations via LLM
+    ```
+  </TabItem>
+</Tabs>
+
+### Disabling Text Rule Processing
+
+If you don't have LLM dependencies installed or want to skip text processing:
+
+<Tabs>
+  <TabItem value="Python" label="Python" default>
+    ```python
+    # Skip text-based rules, generate only predefined and explicit rules
+    rules = generator.generate_rules_from_contract(
+        contract_file="contract.yaml",
+        process_text_rules=False  # Skip LLM processing
+    )
+
+    # Text expectations will be logged as warnings and skipped
+    ```
+  </TabItem>
+</Tabs>
+
+## Complete Workflow Example
```

Comment:
let's avoid the word "Workflow" as it may confuse users, that this is supported using workflows/jobs